server/util.go

```diff
@@ -354,3 +355,47 @@ func dumpTextValue(colInfo *ColumnInfo, value types.Datum) ([]byte, error) {
 		return nil, errInvalidType.Gen("invalid type %v", value.Kind())
 	}
 }
+
+func dumpDatumToBinary(alloc arena.Allocator, column *ColumnInfo, val types.Datum) ([]byte, error) {
```

Comment:
This is used only in x-protocol. Consider moving it into xprotocol package.

Reply:
done.